mantidproject/mantid
# Sqw Presenter is no longer QObject and View is passed to constructor

### Description of work
This PR removes the Qt dependency of the IndirectDataManipulationSqwTab, which is a presenter. Removing this dependency simplifies the code slightly and makes the presenter easier to test due to the better separation of concerns. To make this separation possible, the presenter is now subscribed to the view using a non-owning raw pointer, and then the view will notify the presenter whenever there is an event.

I have created an interface class for the View and Presenter to make this new pattern possible.

The View is now also created outside of the presenter, and passed as an argument to the presenter constructor. This will make it possible to experiment with different Views (the things we see), without needing to change the backend code (presenter and model). We will be able to experiment by creating Views which conform to the same interface as the `ISqwView` and then passing it into the constructor of the presenter.

Part of #36477

**This does not require release notes because it is internal maintenance**

### To test:
Open Inelastic Data Manipulation
Go to the Sqw tab
Load `ExternalData/Testing/Data/UnitTest/iris26176_graphite002_red.nxs` as sample
Click run
See that the plot options at the bottom are enabled
Open the slice Viewer from the plotting options at the bottom

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
